File: app/src/androidTest/java/com/orgzly/android/util/QueryUtilsTest.java
package com.orgzly.android.util;

import android.net.Uri;

import com.orgzly.android.query.Condition;
import com.orgzly.android.query.Query;
import com.orgzly.android.query.user.DottedQueryParser;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.junit.runners.Parameterized;

import java.util.Arrays;

import static org.junit.Assert.assertEquals;

@RunWith(value = Parameterized.class)
public class QueryUtilsTest {

    private final String query;
    private final String expectedBookName;

    public QueryUtilsTest(String query, String expectedBookName) {
        this.query = query;
        this.expectedBookName = expectedBookName;
    }

    @Parameterized.Parameters(name = "{index}: query {0} should return book name {1}")
    public static Iterable<Object[]> data() {
        return Arrays.asList(new Object[][]{
                {"b.foo", "foo"},
                {"b.foo b.bar", "foo"},
                {"", null}
        });
    }

    @Test
    public void testExtractFirstBookNameFromQuery() throws Exception {
        Condition condition = new DottedQueryParser().parse(query).getCondition();
        String result = QueryUtils.extractFirstBookNameFromQuery(condition);

        assertEquals(expectedBookName, result);
    }
}

File: app/src/androidTest/java/com/orgzly/android/query/QueryUtilsTest.kt
package com.orgzly.android.query

import com.orgzly.android.query.user.DottedQueryParser
import org.junit.Assert.assertEquals
import org.junit.Test
import org.junit.runner.RunWith
import org.junit.runners.Parameterized

@RunWith(value = Parameterized::class)
class QueryUtilsTest(private val param: Parameter) {

    data class Parameter(val query: String, val bookName: String?)

    companion object {
        @JvmStatic @Parameterized.Parameters(name = "{index}: query {0} should return book name {1}")
        fun data(): Collection<Parameter> {
            return listOf(
                    Parameter("b.foo", "foo"),
                    Parameter("b.foo b.bar", "foo"),
                    Parameter("foo or b.bar", "bar"),
                    Parameter("", null)
            )
        }
    }

    @Test
    fun testExtractFirstBookNameFromQuery() {
        val condition = DottedQueryParser().parse(param.query).condition
        val result = QueryUtils.extractFirstBookNameFromQuery(condition)
        assertEquals(param.bookName, result)
    }
}